core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -741,6 +758,40 @@ object Doc {
   def split(str: String, pat: Regex = Doc.splitWhitespace, sep: Doc = Doc.lineOrSpace): Doc =
     foldDocs(pat.pattern.split(str, -1).map(Doc.text))((x, y) => x + (sep + y))
 
+  /**
+   * When flattening fails, we still need documents to satisfy
```

Comment:
maybe we should just drop this invariant and try to get as right associated as is cheap to do, but if we have to do O(N) work, we shouldn't do it.

What do you think?

Reply:
Ha, fine with me.  I suppose there can be bad cases where the caller gives you a left-associated doc which is unflattenable, and as such gets copied to both sides of the `Union` by `fill`.  Then when doing `fits` you may be searching through the stack of `Concat`s for awhile before deciding to go right, then see that nasty term again on the right.   This is all very hypothetical though.  How about we just drop it and if a user has performance issues we deal with it then?  We could for example provide an expesive-ish flattener they could use to right-associate their documents before calling `fill`.